h2/src/main/org/h2/bnf/context/DbContextRule.java

```diff
@@ -78,12 +78,18 @@ public boolean autoComplete(Sentence sentence) {
             DbSchema bestSchema = null;
             for (DbSchema schema: schemas) {
                 String name = StringUtils.toUpperEnglish(schema.name);
+                String quotedName = "\"" + schema.quotedName + "\"";
```

Comment:
`schema.quotedName` can be either quoted or unquoted, you shouldn't try to quote it again.

It is actually a normalized name (how it should be inserted into SQL), so you can rename this field and introduce a new one with a real always quoted name (constructed with `StringUtils.quoteIdentifier()`).

Reply:
Thank you very much for feedback and advice. 
I have fixed that and also added tests accordingly for identifiers like  `"quoted schema"` and `"quoted tablename"`.